packages/assets-controllers/src/token-prices-service/abstract-token-prices-service.ts

```diff
@@ -5,7 +5,7 @@ import type { Hex } from '@metamask/utils';
  */
 export type TokenPrice<TokenAddress extends Hex, Currency extends string> = {
   tokenAddress: TokenAddress;
-  value: number;
+  value: number | undefined;
```

Comment:
This is the part I think we should avoid. To me it doesn't make sense that a token price would have a `value` of undefined. You can't do anything meaningful with such an object. It's sort of like having a car without a steering wheel 😅. My thought was that if we couldn't find a price that matches the chain ID or currency in the response we just don't include the token price object at all. That would mean that if you requested prices for multiple token addresses, not all addresses would show up in the object you get back. Hence the change to the return type to use Partial. Does that make sense?